spec/support/test_setup.rb

```diff
@@ -0,0 +1,90 @@
+class TestSetup
+  # Class that instantiates standard set of test data for specs.
+
+  @@recruitment_cycles = {
+    current_cycle: -> { FactoryBot.find_or_create :recruitment_cycle },
+    next_cycle: -> { FactoryBot.find_or_create :recruitment_cycle, :next },
+  }
+
+  @@courses = {
+    unpublished_course_with_primary_maths: ->(attributes = {}) do
+      FactoryBot.find_or_create(:course, :unpublished_with_primary_maths, attributes)
+    end,
+    unpublished_course_with_no_location_or_enrichment: ->(attributes = {}) do
+      FactoryBot.find_or_create(:course, :unpublished_with_primary_maths, {
+        site_statuses: [],
+        enrichments: [],
+      }.merge(attributes))
+    end,
+    unpublished_course_with_draft_enrichment: ->(attributes = {}) do
+      FactoryBot.find_or_create(:course, :unpublished_with_primary_maths, attributes)
+    end,
+    unpublished_fee_type_based_course_with_invalid_enrichment: ->(attributes = {}) do
+      FactoryBot.find_or_create(:course, :fee_type_based, :unpublished_with_primary_maths, {
+        enrichments: [FactoryBot.build(:course_enrichment, :without_content)],
+      }.merge(attributes))
+    end,
+  }
+
+  # Used when not instantiating a TestSetup object in a before block.
+  # This will create a new fixture, using 'find_or_create' for teams and
+  # users and 'create' for cases. This is used to have a standardised set of
+  # recruitment cycles to work with, but without requiring
+  # pre-instantiation.
+
+  def self.method_missing(method_name, *args)
+    if @@recruitment_cycles&.key?(method_name)
+      @@recruitment_cycles[method_name].call
+    elsif @@courses&.key?(method_name)
+      @@courses[method_name].call
+    else
+      super
+    end
+  end
+
+  def self.respond_to_missing?(method_name, include_private = false)
```

Comment:
I'm assuming the comment above the above method applies to this too. Might be worth documenting them individually too as they're pretty hard to parse for someone like me.